lib/pacemaker/libpacemaker_private.h

```diff
@@ -25,8 +25,21 @@ struct resource_alloc_functions_s {
     gboolean(*create_probe) (pe_resource_t *, pe_node_t *, pe_action_t *, gboolean, pe_working_set_t *);
     void (*internal_constraints) (pe_resource_t *, pe_working_set_t *);
 
+    /*!
+     * \internal
+     * \brief Apply a colocation's score to node weights or resource priority
+     *
+     * Given a colocation constraint, apply its score to the dependent's
+     * allowed node weights (if we are still placing resources) or priority (if
+     * we are choosing promotable clone instance roles).
+     *
+     * \param[in] dependent      Dependent resource in colocation
+     * \param[in] primary        Primary resource in colocation
+     * \param[in] colocation     Colocation constraint to apply
+     * \param[in] for_dependent  true if called on behalf of dependent
+     */
     void (*apply_coloc_score) (pe_resource_t *dependent, pe_resource_t *primary,
-                               pcmk__colocation_t *constraint,
+                               pcmk__colocation_t *colocation,
```

Comment:
We should do this in the variant-specific prototypes in libpacemaker-private.h too.

Reply:
Whoops, done